src/commands/synthetics/tunnel/tunnel.ts

```diff
@@ -30,24 +28,30 @@ export interface TunnelInfo {
   privateKey: string
 }
 
+export interface TunnelReporter {
+  error(message: string): void
+  log(message: string): void
+  warn(message: string): void
+}
+
 export class Tunnel {
-  private connected = false
-  private forwardedSockets: Set<Socket> = new Set()
   private FORWARDING_TIMEOUT = 40000 as const
-  private log: (message: string) => void
-  private logError: (message: string) => void
-  private logWarning: (message: string) => void
-  private multiplexer?: Multiplexer
+
+  private sshConfig: ServerConfig
   private privateKey: string
   private publicKey: ParsedKey
-  private sshConfig: ServerConfig
-  private ws: WebSocket
 
-  constructor(private url: string, private testIDs: string[], proxy: ProxyConfiguration, reporter: MainReporter) {
-    this.log = (message: string) => reporter.log(`[${chalk.bold.blue('Tunnel')}] ${message}\n`)
-    this.logError = (message: string) => reporter.error(`[${chalk.bold.red('Tunnel')}] ${message}\n`)
-    this.logWarning = (message: string) => reporter.log(`[${chalk.bold.yellow('Tunnel')}] ${message}\n`)
+  private connected = false
+  private ws: WebSocket
+  private multiplexer?: Multiplexer
+  private forwardedSockets: Set<Socket> = new Set()
 
+  constructor(
+    private url: string,
+    private testIDs: string[],
+    proxyAgent?: ReturnType<typeof ProxyAgent>,
```

Comment:
Why is `ReturnType<typeof ...>` needed here ?

Reply:
`ProxyAgent` is actually the constructor and the `proxy-agent` doesn't seem to expose the actual `ProxyAgent` type sadly